modules/ingestor/src/endpoints/advisory.rs

```diff
@@ -13,85 +19,104 @@ use trustify_module_graph::graph::Graph;
         ("advisory_format" = String, Path, description = "Format of the submitted advisory document (`casf`, `osv`, ...)"),
     ),
     responses(
-        (status = 200, description = "Upload a file"),
-        (status = 400, description = "The file could not be parsed as an advisory document"),
+        (status = 201, description = "Upload a file"),
+        (status = 400, description = "The file could not be parsed as an advisory"),
     )
 )]
-#[post("/advisories/{advisory_format}")]
+#[post("/advisories")]
 /// Upload a new advisory
 pub async fn upload_advisory(
-    graph: web::Data<Graph>,
-    req: HttpRequest,
-    path: web::Path<String>,
+    service: web::Data<IngestorService>,
     payload: web::Payload,
+    web::Query(UploadAdvisoryQuery { location, format }): web::Query<UploadAdvisoryQuery>,
 ) -> Result<impl Responder, Error> {
-    let advisory_format = path.into_inner().to_lowercase();
-
-    let payload_bytes = payload.to_bytes().await?;
-    let payload = BufReader::new(payload_bytes.as_ref());
-
-    if advisory_format == "csaf" {
-        let loader = CsafLoader::new(&graph);
-
-        let advisory_id = loader
-            .load(req.path(), payload)
-            .await
-            .map_err(anyhow::Error::new)?;
-        Ok(HttpResponse::Created().json(advisory_id))
-    } else if advisory_format == "osv" {
-        let loader = OsvLoader::new(&graph);
-
-        let advisory_id = loader
-            .load(req.path(), payload)
-            .await
-            .map_err(anyhow::Error::new)?;
-        Ok(HttpResponse::Created().json(advisory_id))
-    } else {
-        Err(Error::BadRequest {
-            msg: "Unsupported advisory format".to_string(),
-            status: StatusCode::EXPECTATION_FAILED,
-        })
-    }
+    let fmt = format
+        .map(|f| Format::from_str(&f))
+        .unwrap_or(Ok(Format::CSAF))?;
```

Comment:
Default to CSAF format? If someone POSTs in a RANDOFORMAT, which is not actually CSAF, then we'll end up with a bunch of "this is invalid CSAF" errors instead of "We don't support RANDOFORMAT" yeah?

Reply:
My logic may be flawed. How should it work?

Reply:
Maybe `.ok_or(Error::InvalidFormat)?` which will turn a None into a throw error, or a Some into the underlying value without unwrap.  Sorta the err-variant of `unwrap_or(...)`

Reply:
I think my logic is correct: I'm mapping and unwrapping an `Option` containing a `Result`. So the `unwrap_or` is returning an `Err` if the format param is invalid. The `Ok(Format::CSAF)` is only returned if `format` is `None`.

See #126